rpmlint/checks/MenuXDGCheck.py

```diff
@@ -77,7 +59,43 @@ def check_file(self, pkg, filename):
                     error_printed = True
             if not error_printed:
                 self.output.add_info('E', pkg, 'invalid-desktopfile', filename)
-        if not is_utf8(f):
-            self.output.add_info('E', pkg, 'non-utf8-desktopfile', filename)
 
         self.parse_desktop_file(pkg, root, f, filename)
+
+    def _handle_parser_error(self, pkg, filename, e):
+        """
+        Determine what to do with a caught configparser error.
+        """
+        # I would love to use switch, however, each warning is printed differently
+        if (isinstance(e, cfgparser.MissingSectionHeaderError)):
+            self.output.add_info('E', pkg, 'desktopfile-missing-header', filename)
+        elif (isinstance(e, cfgparser.DuplicateSectionError)):
+            self.output.add_info('E', pkg, 'desktopfile-duplicate-section', filename,
+                                 '[%s]' % e.section)
+        elif (isinstance(e, cfgparser.DuplicateOptionError)):
+            self.output.add_info('E', pkg, 'desktopfile-duplicate-option', filename,
+                                 '[%s]/%s' % (e.section, e.option))
+        else:
+            self.output.add_info('E', pkg, 'invalid-desktopfile', filename,
+                                 e.message.partition(':')[0])
+
+    def _has_binary(self, root, cfp, binary):
+        """
+        Check whether there is a binarry assigned to the desktop file.
+
+        Needs configparser instance, it is assumed to be called in parse_desktop_file.
+        """
+        if cfp.has_option('Desktop Entry', 'Exec'):
+            binary.append(cfp.get('Desktop Entry', 'Exec').partition(' ')[0])
+        # If there is no binary mentioned it is OK
+        if not binary[0]:
```

Comment:
why are you checking just binary0, if it uses multiple binaries should't it check more? or what is the iterator for?

Reply:
well, yeah, this is uggly...